wallet/src/Pos/Wallet/Web/Methods/Backup.hs

```diff
@@ -52,18 +53,18 @@ restoreWalletFromBackup WalletBackup {..} = do
                 defaultAccAddrIdx = DeterminedSeed firstHardened
 
             addSecretKey wbSecretKey
-            -- If there are no existing accounts, then create one
             if null accList
                 then do
                     let accMeta = CAccountMeta { caName = "Initial account" }
                         accInit = CAccountInit { caInitWId = wId, caInitMeta = accMeta }
                     () <$ L.newAccountIncludeUnready True defaultAccAddrIdx emptyPassphrase accInit
-                else for_ accList $ \(idx, meta) -> do
-                    let aIdx = fromInteger $ fromIntegral idx
-                        seedGen = DeterminedSeed aIdx
-                    accId <- genUniqueAccountId seedGen wId
-                    createAccount accId meta
-
+                else do
+                    for_ accList $ \(idx, meta) -> do
+                        ws <- getWalletSnapshot db
```

Comment:
Yep, here is necessary to re-read the state at each pass of the loop as we need the updated snapshot to be able to compute the next random index whilst avoiding collisions.